src/main/java/org/gridsuite/study/server/service/StudyService.java

```diff
@@ -321,32 +322,69 @@ public RootNetworkCreationRequestInfos createRootNetworkRequest(UUID studyUuid,
     @Transactional
     public void createRootNetwork(@NonNull UUID studyUuid, @NonNull RootNetworkInfos rootNetworkInfos) {
         StudyEntity studyEntity = studyRepository.findById(studyUuid).orElseThrow(() -> new StudyException(STUDY_NOT_FOUND));
-        Optional<RootNetworkCreationRequestEntity> rootNetworkCreationRequestEntityOpt = rootNetworkService.getCreationRequest(rootNetworkInfos.getId());
+        Optional<RootNetworkRequestEntity> rootNetworkCreationRequestEntityOpt = rootNetworkService.getRootNetworkRequest(rootNetworkInfos.getId());
         if (rootNetworkCreationRequestEntityOpt.isPresent()) {
             rootNetworkInfos.setName(rootNetworkCreationRequestEntityOpt.get().getName());
             rootNetworkInfos.setTag(rootNetworkCreationRequestEntityOpt.get().getTag());
             rootNetworkService.createRootNetwork(studyEntity, rootNetworkInfos);
-            rootNetworkService.deleteCreationRequest(rootNetworkCreationRequestEntityOpt.get());
+            rootNetworkService.deleteRootNetworkRequest(rootNetworkCreationRequestEntityOpt.get());
         } else {
             rootNetworkService.deleteRootNetworks(studyEntity, List.of(rootNetworkInfos));
         }
         notificationService.emitRootNetworksUpdated(studyUuid);
     }
 
-    @Transactional(readOnly = true)
-    public void updateNetworkRequest(UUID studyUuid, UUID rootNetworkUuid, UUID caseUuid, String caseFormat, Map<String, Object> importParameters, String userId) {
+    private void updateRootNetworkBasicInfos(UUID studyUuid, UUID rootNetworkUuid, String name, String tag) {
+        modifyRootNetwork(studyUuid, rootNetworkUuid, RootNetworkInfos.builder()
+                .id(rootNetworkUuid)
+                .name(name)
+                .tag(tag)
+                .build(), false);
+    }
+
+    @Transactional
+    public void updateRootNetworkRequest(UUID studyUuid, UUID rootNetworkUuid, UUID caseUuid, String name, String tag, String caseFormat, Map<String, Object> importParameters, String userId) {
+        rootNetworkService.assertCanModifyRootNetwork(studyUuid, rootNetworkUuid, name, tag);
+        StudyEntity studyEntity = studyRepository.findById(studyUuid).orElseThrow(() -> new StudyException(STUDY_NOT_FOUND));
```

Comment:
Move this entity selection into the `caseUuid != null` case you don't need it otherwise

Reply:
Will be discuss for another PR